src/net.cpp

```diff
@@ -2265,8 +2275,10 @@ void CNode::BeginMessage(const char* pszCommand) EXCLUSIVE_LOCK_FUNCTION(cs_vSen
 {
     ENTER_CRITICAL_SECTION(cs_vSend);
     assert(ssSend.size() == 0);
+    assert(strSendCommand.empty());
```

Comment:
What is this change for?

Reply:
It's asserting that a new send is not started before an existing send is finished. The way Bitcoin Core implemented message sending, each peer has its own message-construction buffer `ssSend`, which is filled as the message is constructed, and then cleared in `EndMessage` as it is queued for sending. I've added the command name alongside it (so we have it for metrics, since we only know the message size, and that we have queued it, in `EndMessage`), so I'm similarly ensuring the same invariant holds.